### ISSUE TITLE
AttributeError: 'module' object has no attribute 'correctiq'

### ISSUE THREAD

**Issue by HarryHydro** [ghostop14/gr-correctiq#3]
Hi: I downloaded the ZIP for this install, unzipped, mkdir,etc. 
I have the three blocks in my list. I created a new project, just the Osmocon, CorrectIQ, and a FFT sink.

This is the error:
AttributeError: 'module' object has no attribute 'correctiq'

hen stack overflows, I guess, OOOOO for days.
I must be missing something.  Any pointers appreciated!
GNU Radio Companion 3.7.9
Thanks!
Harry

**Comment by HarryHydro** [comment]
More importantly, I should have mentioned I'm running this on Ubuntu 16.4.2

**Comment by ghostop14** [comment]
Hi HarryHydro,

I've been running on Ubuntu 16.04.2 too so that probably isn't it.  The GNURadio in the repo (the 3.7.9) is a little old.  I'd actually suggest removing it and installing it from pybombs.  That'll usually pick up all the prerequisites and I know that'll work correctly on Ubuntu 16.04.  

If you want to stick with 3.7.9 you could try a couple things.  If it can't find the module typically either SWIG didn't produce the appropriately wrapped module (look for swig errors during build), or an ldconfig can fix it, but I'd lean towards a swig issue to start.  You can also try 'apt install swig3.0' if it's not already installed (installing from repo may not have installed the latest version), redo the correctiq build process (rm build, cmake again, etc.) and see if you get any SWIG errors now during build.  If it's clean, make install and ldconfig and go back into gnuradio and see if it works okay.  Let me know if any of those do the trick.

**Comment by HarryHydro** [comment]
blockquote, div.yahoo_quoted { margin-left: 0 !important; border-left:1px #715FFA solid !important; padding-left:1ex !important; background-color:white !important; } Thanks!  I'll give it a try.  I found another module with a magnitude and phase. Stock module I think. It in itself bulled the spike with sliders, but it seemed to create negative images on top of the fft. I used a multiplier, and a separate adder and was able to null the spike, but AGC seemed to bring it back. Might be version problems. ThanksHarry


Sent from Yahoo Mail for iPhone


On Sunday, June 4, 2017, 9:35 AM, ghostop14 <notifications@github.com> wrote:


Hi HarryHydro,

I've been running on Ubuntu 16.04.2 too so that probably isn't it. The GNURadio in the repo (the 3.7.9) is a little old. I'd actually suggest removing it and installing it from pybombs. That'll usually pick up all the prerequisites and I know that'll work correctly on Ubuntu 16.04.

If you want to stick with 3.7.9 you could try a couple things. If it can't find the module typically either SWIG didn't produce the appropriately wrapped module (look for swig errors during build), or an ldconfig can fix it, but I'd lean towards a swig issue to start. You can also try 'apt install swig3.0' if it's not already installed (installing from repo may not have installed the latest version), redo the correctiq build process (rm build, cmake again, etc.) and see if you get any SWIG errors now during build. If it's clean, make install and ldconfig and go back into gnuradio and see if it works okay. Let me know if any of those do the trick.

**Comment by HarryHydro** [comment]
Brother! You da Man! I installed swig the way you said, reinstalled the CorrectIQ, and it's working!
Before I did this though, I uninstalled gnuradio.Searched for a newer version on the GNURadio site.  Found links to newer versions.  All corrupt security certificates Can't go there.Downloaded a different browser. That went there. Downloaded a .11 version. Can't install it. Nowhere to be found.Tried to download a 'torrent'. Porno pop-ups! WTF? Quit that.
Found a version specifically for me, AMD64, Ubuntu, etc. ISO file. Downloaded.
Double-clicked that, opened into a bunch of folders and/or symbolic links. 
Tried copying these to a USB stick. Booted with it. boot: every 10 seconds. no kernel.  
downloaded unetbootin something iso burner. binary file. Fiddled with that to get that working. Placement.'Burned the iso to the USB drive. Booted with it, unetbootin prompt?Burnt again. Same thing. No kernel either. Gave up on that.Downloaded the swig3.0.  That worked! Tried to install the correctIq. Failed. Can't find GRURadio. I deleted it earlier. 
Reinstalled GNURadio via apt install gnuradio. InstalledInstalled correctIQ again. It works!ThanksHarry
